src/rhsmlib/facts/network.py

```diff
@@ -0,0 +1,176 @@
+
+
+class NetworkCollector(collector.FactsCollector):
+    def __init__(
+        self,
+        arch: str = None,
+        prefix: str = None,
+        testing: bool = None,
+        collected_hw_info: Dict[str, Union[str, int, bool, None]] = None,
+    ):
+        super().__init__(arch=arch, prefix=prefix, testing=testing, collected_hw_info=None)
+
+        self.hardware_methods: List[Callable] = [
+            self.get_network,
+            self.get_interfaces,
+        ]
+
+    def _query_ip_command(self) -> List[dict]:
+        """Call system's 'ip' command and return its value as a dictionary."""
+        output = subprocess.run(["ip", "--json", "address"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if output.stderr != b"":
+            log.error(f"Could not query 'ip' for network facts: {output.stderr}")
+            return []
+        stdout: str = output.stdout.decode("utf-8")
+        return json.loads(stdout)
+
+    def _extract_address_list(self, family: Literal["inet", "inet6"], data: List[dict]) -> List[str]:
+        """Extract IPv4 or IPv6 addresses from 'ip' output.
+
+        The list excludes loopback and link-local addresses.
+        """
+        result: List[str] = []
+        for interface in data:
+            if "LOOPBACK" in interface["flags"]:
+                continue
+            result += [
+                address["local"]
+                for address in interface["addr_info"]
+                if address["family"] == family and address["scope"] != "link"
+            ]
+        return result
+
+    def get_network(self) -> dict:
+        """Get general network facts.
+
+        Hostname, FQDN and a list of IPv4/IPv6 addresses get collected here.
+
+        Resulting facts have 'network.' prefix.
+        """
+        # These socket functions work even with no network available
+        data = self._query_ip_command()
+
+        result: Dict[str, str] = {
+            "network.hostname": socket.gethostname(),
+            "network.fqdn": socket.getfqdn(),
+            "network.ipv4_address": ", ".join(self._extract_address_list("inet", data)),
+            "network.ipv6_address": ", ".join(self._extract_address_list("inet6", data)),
+        }
+        return result
+
+    def get_interfaces(self) -> dict:
+        """Get detailed network interface facts.
+
+        Interface names, IPv4/IPv6 addresses and masks get collected here.
+
+        Resulting facts have 'net.' prefix.
+        """
+        data = self._query_ip_command()
+        result: Dict[str, Union[str, int]] = {}
+
+        for interface in data:
+            prefix: str = f"net.interface.{interface['ifname']}"
+
+            # MAC address
+            if "LOOPBACK" in interface["flags"] or "NOARP" in interface["flags"]:
+                # Loopback has a MAC address of '00:00:00:00:00:00'.
+                # Tunnels have their MAC randomized every time, see BZ#838123.
+                pass
+            else:
```

Comment:
instead of
```python
  if ...:
    pass
  else:
    ...
```
i'd do:
```python
  if ...:
    continue
  ...
```
which i find easier to read and understand that the `if` block terminates earlier every `for` iteration, and also makes it possible to reduce the indentation level of the current `else` branch by one; in addition, `prefix` can be created after the `if`, ie only when needed